intake_xarray/image.py

```diff
@@ -0,0 +1,265 @@
+    """
+    import numpy as np
+    from xarray import DataArray
+
+    if not imread:
+        from skimage.io import imread
+
+    with file as f:
+        array = imread(f)
+    if preprocess:
+        array = preprocess(array)
+
+    ny, nx = array.shape[:2]
+    coords = {'y': np.arange(ny),
+              'x': np.arange(nx)}
+    dims = ('y', 'x')
+
+    if len(array.shape) == 3:
+        nband = array.shape[2]
+        coords['band'] = np.arange(nband)
+        dims += ('band',)
+
+    return DataArray(array, coords=coords, dims=dims).chunk(chunks=chunks)
+
+
+def multireader(files, chunks, concat_dim, **kwargs):
+    import numpy as np
+    from xarray import DataArray
+
+    dask_array = dask_imread(files, **kwargs)
+
+    ny, nx = dask_array.shape[1:3]
+    coords = {'y': np.arange(ny),
+              'x': np.arange(nx)}
+    if isinstance(concat_dim, list):
+        dims = ('dim_0', 'y', 'x')
+    else:
+        dims = (concat_dim, 'y', 'x')
+
+    if len(dask_array.shape) == 4:
+        nband = dask_array.shape[3]
+        coords['band'] = np.arange(nband)
+        dims += ('band',)
+
+    return DataArray(dask_array, coords=coords, dims=dims).chunk(chunks=chunks)
+
+
+class ImageSource(DataSourceMixin, PatternMixin):
+    """Open a xarray dataset from image files.
+
+    This creates an xarray.DataArray or an xarray.Dataset.
+    See http://scikit-image.org/docs/dev/api/skimage.io.html#skimage.io.imread
+    for the file formats supported, and
+    http://docs.dask.org/en/latest/array-api.html#dask.array.image.imread
+    for possible extra arguments.
+
+    NOTE: Although ``skimage.io.imread`` is used by default, any reader function which
+    accepts a filename and outputs a numpy array can be used instead.
+
+    Parameters
+    ----------
+    urlpath : str or iterable, location of data
+        May be a local path, or remote path if including a protocol specifier
+        such as ``'s3://'``. May include glob wildcards or format pattern strings.
+        Must be a format supported by ``skimage.io.imread`` or user-supplied ``imread``
+        Some examples:
+            - ``{{ CATALOG_DIR }}data/RGB.tif``
+            - ``s3://data/*.jpeg``
+            - ``https://example.com/image.png`
+            - ``s3://data/Images/{{ landuse }}/{{ landuse }}{{ '%02d' % id }}.tif``
+    chunks: int or dict
+        Chunks is used to load the new dataset into dask
+        arrays. ``chunks={}`` loads the dataset with dask using a single
+        chunk for all arrays.
+    path_as_pattern: bool or str, optional
+        Whether to treat the path as a pattern (ie. ``data_{field}.tif``)
+        and create new coodinates in the output corresponding to pattern
+        fields. If str, is treated as pattern to match on. Default is True.
+    concat_dim : str or iterable
+        Dimension over which to concatenate. If iterable, all fields must be
+        part of the the pattern.
+    xarray_kwargs : dict, optional
```

Comment:
There were only two? Or are there additional kwargs that might be important here?